source/backend/vulkan/execution/VulkanMatrixMultier4x4.cpp

```diff
@@ -82,6 +82,10 @@ void VulkanMatrixMultier4x4::prepare(int e, std::shared_ptr<VulkanImage> dst, st
         mDest->release();
     }
 
+    commandBuffer->barrierImageIfNeeded(mDest.get(), VK_IMAGE_LAYOUT_GENERAL);
+    commandBuffer->barrierImageIfNeeded(mSource.get(), VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL);
+    commandBuffer->barrierImageIfNeeded(mKernel.get(), VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL);
```

Comment:
prepare and compute both barrier may be not needed?

Reply:
@jxt1234 Yes, my code was quite defensive. Maybe we can remove barrier-related code in `compute()`

Reply:
```c++
void VulkanMatrixMultier4x4::compute(const VulkanCommandPool::Buffer* commandBuffer) const {
    mPipeline->bind(commandBuffer->get(), mDescriptorSet->get());
    vkCmdDispatch(commandBuffer->get(), UP_DIV(mOutputWidth, 8), UP_DIV(mOutputHeight / 4, 8), mDepth);
}
```
I've checked that this modified code also does not cause validation errors. I'll push it.